modules/oe_webtools_wtag/src/Element/Wtag.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_webtools_wtag\Element;
+
+use Drupal\Component\Serialization\Json;
+use Drupal\Core\Entity\Element\EntityAutocomplete;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Render\Markup;
+use Drupal\rdf_skos\Entity\ConceptInterface;
+
+/**
+ * Wtag form element integrating the Webtools Wtag service.
+ *
+ * It extends the original EntityAutocomplete form element and can reference
+ * only SKOS Concept entities.
+ *
+ * @FormElement("oe_webtools_wtag")
+ */
+class Wtag extends EntityAutocomplete {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getInfo() {
+    $info = parent::getInfo();
+
+    $class = static::class;
+    $info['#process'][] = [$class, 'processWtag'];
+
+    // Apply default form element properties.
+    $info['#target_type'] = 'skos_concept';
+    $info['#tags'] = TRUE;
+    $info['#theme'] = 'oe_webtools_wtag';
+    $info['#modal_title'] = $this->t('Wtag');
+    $info['#modal_description'] = $this->t('Use the search field or the tree view to find and add one or more tags to describe the subject of your page.');
+
+    return $info;
+  }
+
+  /**
+   * Process callback for the element.
+   *
+   * @param array $element
+   *   The form element to process.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The current state of the form.
+   * @param array $complete_form
+   *   The complete form structure.
+   *
+   * @return array
+   *   The form element.
+   */
+  public static function processWtag(array &$element, FormStateInterface $form_state, array &$complete_form) {
+    $wtag_json = [
+      'service' => 'wtag',
+      'target' => '#' . $element['#id'],
+      'title' => $element['#modal_title'],
+      'description' => $element['#modal_description'],
+    ];
+
+    $element['#wtag'] = Markup::create(Json::encode($wtag_json));
+    $element['#attached']['library'][] = 'oe_webtools/drupal.webtools-smartloader';
+    unset($element['#attributes']['data-autocomplete-path']);
+    $element['#attributes']['maxlength'] = '100000000';
```

Comment:
Are both maxlengths needed?

Reply:
Well, I think so, don't remember 100%. One is for the HTML element itself, the other is checked by a validator somewhere..